src/BBT.StructureTools/BBT.StructureTools.csproj

```diff
@@ -22,7 +22,10 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.9.3">
+    <PackageReference Include="BBT.Maybe" Version="2.0.0" />
+    <PackageReference Include="BBT.StrategyPattern" Version="1.0.0" />
+    <PackageReference Include="FluentAssertions" Version="5.9.0" />
```

Comment:
@marco-bertschi Making this a private asset doesn't help (it actually makes it worse). Since you use the library it needs to be there at runtime and the package needs to have a dependency on FluentAssertions. Which introduces a tight coupling and forces every consumer to also deploy FluentAssertions. We should not use FluentAssertions in the `BBT.StructureTools` assembly at all, but instead use some internal classes for argument validation.

Reply:
You can use something like [this](https://github.com/cake-contrib/Cake.Issues/blob/develop/src/Cake.Issues/IssuesArgumentChecks.cs) for argument checking (but make it internal)

Reply:
@pascalberger i'm not happy to generate code that is perfectly done by a tested reference only to remove the dependency. I understand your point, but we duplicate it in each GitHub repo. I think your argument is only legitim because it is a very base library.

Reply:
@Speeedy01 I didn't say that you never should take dependencies and duplicate code instead. As always it depends.
 
But you should always carefully consider costs/benefit of taking a dependency. The project consuming the dependency and the consumed dependency will have an impact on this consideration. For example if you introduce a dependency on FluentAssertions in a test project there won't be any downstream consumers affected, the same if you take this dependency in an application. Contrary if you introduce a dependency in a library, you basically force this dependency in the specific major version to all consumers and their consumers (technically you not enforcing the major version in all cases, since there are binding redirects, but thats another story). If for example BBT.Maybe has taken a dependency on FluentAssertions 4.x, while BBT.StrategyPattern has taken a dependency on FluentAssertions 5.x, this could lead to the situation that you won't be able to consume these two libraries together here. Also if the decision while writing BBT.Maybe was to use library A for argument validation, and in BBT.StrategyPattern library B, this would now add both dependencies to this library. While you can argument that you have full control over these libraries, that's something which is only true in this case. But it won't help you in a situation where you for example want to take a dependency on a really helpful librariy in an test assembly of an application consuming BBT.StructureTools, which requires FluentAssertions 42.x, which is incompatible with the version used here. The decision to take a dependency to FluentAssertions here would now block your capability to make use of this useful library. Updating the dependency here and bringing it to the downstream application might not always be possible as this might introduce breaking runtime changes, while you only wanted to use a library in a testing project.
Having these kind of risk taking a dependency on FluentAssertions, have now a look at the costs to not take the dependency. As it is only used for argument checking, this is something which can be written in a handful lines of code, and which is code without a big complexity and risk. 

In another project or with another dependency the analysis of risk and costs obviously would be different.

So the risk in this situation IMHO cleary outweights the costs by factors and therefore should not be taken (which btw was also something which was not done in BBT.Maybe and BBT.StrategyPattern).

Reply:
@pascalberger I'm with you, we don't win a lot when we're using FluentAssertions here - Apart from that it is mostly null checks, and a library isn't giving us any greater advantage. If we mess a null check method up we deserve to be publicly shamed. 
@Speeedy01 If you can have an in-depth look that'd be great. Also, do you still have the regex around you've used to replace the other null checks? I'd like to have it 💯

Reply:
@pascalberger I'm with you, we don't win a lot when we're using FluentAssertions here - Apart from that it is mostly null checks, and a library isn't giving us any greater advantage. If we mess a null check method up we deserve to be publicly shamed. 
@Speeedy01 If you can have an in-depth look that'd be great. Also, do you still have the regex around you've used to replace the other null checks? I'd like to have it 💯

Reply:
@marco-bertschi 
Check the regular expresseion in visual Studio search and replace:
Search:
`([\n\r])(\s*)(.*)(.Should\(\))(.NotBeNull\(\))(;)`
Replace:
'$1$2MyOwnPower.YourOwnNotNullPower($3)'

The doller is the content of the group.
And my cheet sheet: https://regexr.com/